File: frontend/src/types/stats.ts
export interface StatsSummary {
  start_date: string;
  end_date: string;
  trip_count: number;
  total_nights: number;
  total_price: number | null;
  avg_price_per_night: number | null;
  avg_star_rating: number | null;
}

export interface MonthlyStat {
  month: string;
  trip_count: number;
  total_nights: number;
  total_price: number | null;
}

export interface StatsByMonthResponse {
  start_date: string;
  end_date: string;
  months: MonthlyStat[];
}

export interface StatsRange {
  start_date?: string;
  end_date?: string;
}

File: backend/app/schemas/stats.py
from datetime import date

from pydantic import BaseModel


class StatsSummary(BaseModel):
    start_date: date
    end_date: date
    trip_count: int
    total_nights: int
    total_price: float | None
    avg_price_per_night: float | None
    avg_star_rating: float | None


class MonthlyStat(BaseModel):
    month: str
    trip_count: int
    total_nights: int
    total_price: float | None


class StatsByMonthResponse(BaseModel):
    start_date: date
    end_date: date
    months: list[MonthlyStat]
